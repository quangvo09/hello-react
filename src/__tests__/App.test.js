import React from 'react';
import { mount, shallow } from 'enzyme';

import AppComponent from '../App';

describe('App', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<AppComponent debug/>)
    expect(component).toMatchSnapshot();
  })
  it('should render correctly with no props', () => {
    const component = shallow(<AppComponent/>)
    expect(component).toMatchSnapshot();
  })
  it('should render banner text correctly with given strings', () => {
    const strings = ['one', 'two'];
    const component = shallow(<AppComponent list={strings} />);
    expect(component).toMatchSnapshot();
  });
  it('should be possible to activate button with Spacebar', () => {
    const component = mount(<AppComponent />);
    component
      .find('button#my-button-one')
      .simulate('keydown', { keyCode: 32 });
    expect(component).toMatchSnapshot();
    component.unmount();
  });
})

const clickFunc = jest.fn();
describe('App interact', () => {
  it('button click should hide component', () => {
    const component = shallow(<AppComponent onClick={clickFunc} />)
    component
      .find('button#my-button-one')
      .simulate('click');
    expect(clickFunc).toHaveBeenCalled();
  })
})

const mockTryGetValue = jest.fn(() => false);
jest.mock('../save-to-storage', () => ({   
  SaveToStorage: jest.fn().mockImplementation(() => ({
    getValue: mockTryGetValue,
  })), 
}));
mockTryGetValue.mockReturnValueOnce(false);

describe('App save to storage', () => {
  it('should set storage on save button click', () => {
    
    const component = shallow(<AppComponent onClick={clickFunc} />)
    component
      .find('button#my-button-two')
      .simulate('click')
      .simulate('click');
    expect(mockTryGetValue).toHaveBeenCalled();
    expect(component).toMatchSnapshot();
    component.unmount();  
  })
})
