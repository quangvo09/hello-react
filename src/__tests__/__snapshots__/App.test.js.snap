// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App save to storage should set storage on save button click 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <button
      id="my-button-one"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      Button1
    </button>
    <button
      id="my-button-two"
      onClick={[Function]}
    >
      Button2
    </button>
  </header>
</div>
`;

exports[`App should be possible to activate button with Spacebar 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <button
        id="my-button-one"
      >
        Button1
      </button>
      <button
        id="my-button-two"
        onClick={[Function]}
      >
        Button2
      </button>
    </header>
  </div>
</App>
`;

exports[`App should render banner text correctly with given strings 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <button
      id="my-button-one"
    >
      Button1
    </button>
    <button
      id="my-button-two"
      onClick={[Function]}
    >
      Button2
    </button>
  </header>
</div>
`;

exports[`App should render correctly in "debug" mode 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <button
      id="my-button-one"
    >
      Button1
    </button>
    <button
      id="my-button-two"
      onClick={[Function]}
    >
      Button2
    </button>
  </header>
</div>
`;

exports[`App should render correctly with no props 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <button
      id="my-button-one"
    >
      Button1
    </button>
    <button
      id="my-button-two"
      onClick={[Function]}
    >
      Button2
    </button>
  </header>
</div>
`;
